src/Raven.Server/Documents/ShardedHandlers/ShardedCommands/ShardedQueryCommand.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Raven.Client;
+using Raven.Client.Documents.Queries;
+using Raven.Client.Exceptions.Documents.Indexes;
+using Raven.Client.Json.Serialization;
+using Raven.Server.Documents.Sharding;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.ShardedHandlers.ShardedCommands;
+
+public class ShardedQueryCommand : ShardedBaseCommand<QueryResult>
+{
+    private readonly string _indexName;
+
+    public ShardedQueryCommand(ShardedRequestHandler handler, BlittableJsonReaderObject content, string indexName) : base(handler, ShardedCommands.Headers.None, content)
+    {
+        _indexName = indexName;
+        Headers[Constants.Headers.MissingIncludes] = "true";
```

Comment:
what if we would introduce always, for all ShardedCommands a header like Sharded : true
And if needed used that. In your case you are not actually making any logic based on this value and it could be substituted by the generic header.

Reply:
done